apps/omg/lib/omg/utxo.ex

```diff
@@ -36,25 +36,7 @@ defmodule OMG.Utxo do
     end
   end
 
-  defguard is_position(blknum, txindex, oindex)
-           when is_integer(blknum) and blknum >= 0 and
-                  is_integer(txindex) and txindex >= 0 and
-                  is_integer(oindex) and oindex >= 0
-
-  @interval elem(OMG.Eth.RootChain.get_child_block_interval(), 1)
-  @doc """
-  Based on the contract parameters determines whether UTXO position provided was created by a deposit
-  """
-  defguard is_deposit(position)
-           when is_tuple(position) and tuple_size(position) == 4 and
-                  is_position(elem(position, 1), elem(position, 2), elem(position, 3)) and
-                  rem(elem(position, 1), @interval) != 0
-
-  defmacrop is_nil_or_binary(binary) do
-    quote do
-      is_binary(unquote(binary)) or is_nil(unquote(binary))
-    end
-  end
+  defguardp is_nil_or_binary(creating_tx_hash) when is_nil(creating_tx_hash) or is_binary(creating_tx_hash)
```

Comment:
looking at where this guard is used I don't understand what it's guarding us from and why? do you know @achiurizo ? what's the purpose of this guard?

Reply:
I think the guard is being overly defensive here. But to minimize the amount of changes, I left the guard in until a follow up PR.

Reply:
ok